### PR TITLE
Check for warnings output during unit tests

### PR DESCRIPTION
**Pull request by gnikonorov**
In the tests, check to make sure that:
- The correct warnings are reported
- The warnings text is correct

Before, we weren't checking but I think we should be since we are setting these warnings in this plugin, and they are not coming from pytest.

No change log as this is trivial

### CODE REVIEW

**Comment by icemac** [comment]
Hm, the tests fail for `pytest < 5.3`. Do you have any idea why?

**Comment by The-Compiler** [comment]
The problem is that the stored outcomes are `{'failed': 1, 'rerun': 3, 'warnings': 1}` but you're checking for `'warning'` (singular). There were various changes to the plurality there, starting with https://github.com/pytest-dev/pytest/pull/5990 - I'm not sure if `warnings` works on all versions though... Probably worth a try?

**Comment by gnikonorov** [comment]
> The problem is that the stored outcomes are `{'failed': 1, 'rerun': 3, 'warnings': 1}` but you're checking for `'warning'` (singular). There were various changes to the plurality there, starting with [pytest-dev/pytest#5990](https://github.com/pytest-dev/pytest/pull/5990) - I'm not sure if `warnings` works on all versions though... Probably worth a try?

Thanks for the tip @The-Compiler, let me try tonight

**Comment by gnikonorov** [comment]
So I realized what I was doing was inherently wrong. Checking general warnings is extremely fragile at best, since pytest itself emits warnings and there's no way of knowing which warnings came from this plugin vs. pytest. Checking to make sure we emitted at least some amount of warnings 'x' also doesn't tell us anything since those warnings could be coming from somewhere else, so the test would give a false sense of security or just confuse people.

That leads me to think there are 2 possible solutions:
- We keep track of the number of warnings emitted by this plugin in the plugin itself, and compare against that. I personally don't think this is the best idea since we're adding code purely for test purposes, and there's no guarantee that people will maintain this convention
- We do what I updated this PR to do, and just check that stdout contains the warning text we want. I think this is best, since it makes no assumption on the warnings pytest emits, and requires no special maintenance.

Thoughts @icemac and @The-Compiler ?

File: test_pytest_rerunfailures.py [hunk 1]
```diff
@@ -1,4 +1,5 @@
 from unittest import mock
+import pkg_resources
```
Comment (icemac) [inline test_pytest_rerunfailures.py:2]: This import is no longer needed.

File: test_pytest_rerunfailures.py [hunk 2]
```diff
@@ -237,13 +238,17 @@ def test_fail():
     result = testdir.runpytest('--reruns', '3',
                                '--reruns-delay', str(delay_time))
 
+    num_warnings = 0
```
Comment (The-Compiler) [inline test_pytest_rerunfailures.py:241]: This could probably be a `has_warnings` or `warning_expected` boolean now?

Comment (gnikonorov) [reply test_pytest_rerunfailures.py:241]: ah yes, this is wrong. sorry for leaving it, let me remove it

Comment (The-Compiler) [reply test_pytest_rerunfailures.py]: The new version looks better, though now you have two identical `if` statements after each other.

Comment (gnikonorov) [reply test_pytest_rerunfailures.py]: sorry just changed it

**Pull request merged**